### PR TITLE
feat: experience

### PR DESCRIPTION
**Pull request by rickklaasboer**
Closes https://github.com/rickklaasboer/wego-overseer/issues/84

image
image

### CODE REVIEW

**Comment by Joehoel** [comment]
Maybe nice to generate an image for the leaderboard to display instead of a code block.

File: src/app/commands/experience/ExperienceResetCommand.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    ModalBuilder,
+    TextInputBuilder,
+    TextInputStyle,
+    ActionRowBuilder,
+    ModalActionRowComponentBuilder,
+} from 'discord.js';
+import {injectable} from 'tsyringe';
+
+@injectable()
+export default class ExperienceAddCommand extends BaseInternalCommand {
```
Comment (rickklaasboer) [inline src/app/commands/experience/ExperienceResetCommand.ts:15]:
```suggestion
export default class ExperienceResetCommand extends BaseInternalCommand {

```

File: src/app/events/experience/LevelUpEvent.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                message.author.id,
+            );
+            const experience = await this.experienceRepository.getExperience(
+                message.guild?.id ?? '',
+                message.author.id,
+            );
+
+            const newLevel = this.experienceService.xpToLevel(experience, true);
+
+            if (newLevel > level) {
```
Comment (rickklaasboer) [inline src/app/events/experience/LevelUpEvent.ts:43]: Should send level up message too

File: src/app/repositories/UserGuildLevelRepository.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import UserGuildLevel from '@/app/entities/UserGuildLevel';
+import {injectable} from 'tsyringe';
+
+@injectable()
+export default class UserGuildLevelRepository {
+    /**
+     * Gets the level of user.
+     */
+    public async getLevel(guildId: string, userId: string): Promise<number> {
```
Comment (rickklaasboer) [inline src/app/repositories/UserGuildLevelRepository.ts:9]: Could probably determine level here if not available

File: src/app/services/ExperienceService.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import {injectable} from 'tsyringe';
+
+@injectable()
+export default class ExperienceService {
+    // Magic value
+    private readonly multiplier = 0.11;
```
Comment (rickklaasboer) [inline src/app/services/ExperienceService.ts:6]: Maybe make configurable

File: src/util/string.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export function capitalize(str: string): string {
```
Comment (rickklaasboer) [inline src/util/string.ts:1]: Could use some jsdoc

**Pull request merged**